operator/apis/lifecycle/v1alpha3/keptntaskdefinition_webhook.go

```diff
@@ -77,20 +77,33 @@ func (r *KeptnTaskDefinition) validateKeptnTaskDefinition() error {
 		allErrs)
 }
 func (r *KeptnTaskDefinition) validateFields() *field.Error {
-
-	if r.Spec.Function == nil && r.Spec.Container == nil {
+	if r.Spec.Function == nil && r.Spec.Container == nil && r.Spec.Python == nil && r.Spec.Deno == nil {
 		return field.Invalid(
 			field.NewPath("spec"),
 			r.Spec,
-			errors.New("Forbidden! Either Function or Container field must be defined").Error(),
+			errors.New("Forbidden! Either Function, Container, Python, or Deno field must be defined").Error(),
 		)
 	}
 
-	if r.Spec.Function != nil && r.Spec.Container != nil {
+	count := 0
+	if r.Spec.Function != nil {
```

Comment:
would it also be possible to extract this piece of code (the count++ handling etc) to a separate helper function, so the code is easier to read ?